### PR TITLE
Cleanup & Header component

### PR DESCRIPTION
**Pull request by Valentin-Dumitrache**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,29 +1,25 @@
...
-## Project setup
+Homefood is a platform for selling and buying home-made food
 
-```
-npm install
-```
+- See the webpage [page](https://Alina-Dumitrache.github.io/homefood-frontend) related to this project.
 
-### Compiles and hot-reloads for development
+# Build With:
```
Comment (alina-dumitrache) [inline README.md]: Typo here and in own template: `Built with`, not `Build With`. Please update in both places to prevent the error from propagating

File: src/App.vue [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
-    <div id="nav">
-      <router-link to="/">Home</router-link> |
-      <router-link to="/about">About</router-link>
-    </div>
+    <app-header />
     <router-view />
   </div>
 </template>
 
 <style lang="scss">
```
Comment (alina-dumitrache) [inline src/App.vue]: Please move the style tag at the end, the standard order for Vue files is template, script, styles.

File: src/App.vue [hunk 3]
```diff
@@ -1,16 +1,15 @@
...
+@import url("https://fonts.googleapis.com/css?family=Roboto:100,400,700&display=swap");
```
Comment (alina-dumitrache) [inline src/App.vue]: TODO: Investigate how else we can include this font (@alina-dumitrache, @george-iuriet)

File: src/App.vue [hunk 4]
```diff
@@ -30,3 +29,11 @@
   }
 }
 </style>
+<script>
+import AppHeader from "@/components/app-header";
+
+export default {
+  name: "App",
```
Comment (alina-dumitrache) [inline src/App.vue]: Use single quotes, why did the linter not work?

File: src/components/app-header.vue [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        <v-icon x-large>mdi-account-circle</v-icon>
+      </router-link>
+    </v-toolbar>
+    <v-divider />
+  </header>
+</template>
+
+<script>
+export default {
+  name: "appHeader"
```
Comment (alina-dumitrache) [inline src/components/app-header.vue]: Use single quotes

File: src/components/app-header.vue [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    <v-toolbar flat>
+      <router-link to="/">
+        <p>homefood</p>
+      </router-link>
+      <v-spacer />
+      <router-link id="becomeACook" v-if="true" to="/signUp">
+        <v-btn outlined>Become a cook</v-btn>
+      </router-link>
+      <router-link id="logIn" v-if="false" to="/login">Log in</router-link>
+      <router-link id="avatarIcone" v-else to="/profile">
```
Comment (alina-dumitrache) [inline src/components/app-header.vue]: typo: `avatarIcone`

File: src/components/app-header.vue [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  </header>
+</template>
+
+<script>
+export default {
+  name: "appHeader"
+};
+</script>
+
+<!-- Add "scoped" attribute to limit CSS to this component only -->
```
Comment (alina-dumitrache) [inline src/components/app-header.vue]: Good, please make sure to add scoped attribute to all style tags, this pattern prevents unwanted style overlaps.

File: src/components/app-header.vue [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+<style scoped lang="scss">
+p {
+  color: orange;
+  font-size: 170%;
```
Comment (alina-dumitrache) [inline src/components/app-header.vue]: Let's discuss typography, we have to define heading types for the project (Title, subtitle etc.) and define re-usable classes for them (if Vuetify does not have pre-defined ones). Hard-coded font-size as percentage is not a desired code pattern.

Comment (alina-dumitrache) [reply src/components/app-header.vue]: See here: https://v15.vuetifyjs.com/en/framework/typography

File: src/router/index.js [hunk 9]
```diff
@@ -1,24 +1,23 @@
...
-    path: "/about",
-    name: "About",
-    // route level code-splitting
-    // this generates a separate chunk (about.[hash].js) for this route
-    // which is lazy-loaded when the route is visited.
-    component: () =>
-      import(/* webpackChunkName: "about" */ "../views/About.vue")
+    path: "/login",
+    name: "login",
+    component: []
```
Comment (alina-dumitrache) [inline src/router/index.js]: Please use `null` instead of empty array here, to be consistent with the expected data type.

File: src/App.vue (hunk 4 above)
Comment (Valentin-Dumitrache) [reply src/App.vue]: Forgot to mention it in package.json, will fix in next commit

File: src/components/app-header.vue (hunk 7 above)
Comment (Valentin-Dumitrache) [reply src/components/app-header.vue]: Right, this was the default, I'll keep it in mind.

File: src/components/app-header.vue (hunk 8 above)
Comment (Valentin-Dumitrache) [reply src/components/app-header.vue]: Understood, makes sense, I will look into it, and come up with a reply.

Comment (Valentin-Dumitrache) [reply src/components/app-header.vue]: Looked into it, but can't properly test, I've added the class on the element, but the stylesheet which contains it doesn't seem to be used. Should I do an additional step?

File: src/components/app-header.vue [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+<script>
+export default {
+  name: "appHeader"
+};
+</script>
+
+<!-- Add "scoped" attribute to limit CSS to this component only -->
+<style scoped lang="scss">
+p {
+  color: orange;
```
Comment (george-iuriet) [inline src/components/app-header.vue]: let's use Vuetify variables, or find alternative for this hardcoded value.

File: src/components/app-header.vue [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+  font-size: 170%;
+  font-weight: bold;
+}
+#logIn {
+  float: right;
```
Comment (george-iuriet) [inline src/components/app-header.vue]: why is this needed?

File: src/components/app-header.vue [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+  name: "appHeader"
+};
+</script>
+
+<!-- Add "scoped" attribute to limit CSS to this component only -->
+<style scoped lang="scss">
+p {
+  color: orange;
+  font-size: 170%;
+  font-weight: bold;
```
Comment (george-iuriet) [inline src/components/app-header.vue]: prefer helper class `font-weight-bold`. More about it [here](https://vuetifyjs.com/en/styles/typography/)

File: src/components/app-header.vue [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+}
+#logIn {
+  float: right;
+}
+#becomeACook {
+  margin-right: 20px;
```
Comment (george-iuriet) [inline src/components/app-header.vue]: prefer Material Design spacing variables, or helper classes. More about it [here](https://vuetifyjs.com/en/styles/spacing/)

File: src/components/app-header.vue [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+<style scoped lang="scss">
+p {
+  color: orange;
+  font-size: 170%;
+  font-weight: bold;
+}
+#logIn {
+  float: right;
+}
+#becomeACook {
```
Comment (george-iuriet) [inline src/components/app-header.vue]: It is a common approach to never use ids. Generally, we don't like one-thing-only approach, as we always aim at reusing our code as much as possible. Although I don't have strong arguments against using them, I would still recommend doing so. I'll leave it up to you.

File: src/components/app-header.vue [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+  margin-right: 20px;
+}
+a {
```
Comment (george-iuriet) [inline src/components/app-header.vue]: I don't think this rule is needed.

File: src/router/index.js [hunk 16]
```diff
@@ -1,24 +1,23 @@
...
   }
+  // {
```
Comment (george-iuriet) [inline src/router/index.js]: you can safely remove this comment

File: src/router/index.js [hunk 17]
```diff
@@ -1,24 +1,23 @@
...
+  //   path: "/about",
+  //   name: "About",
+  //   // route level code-splitting
+  //   // this generates a separate chunk (about.[hash].js) for this route
+  //   // which is lazy-loaded when the route is visited.
+  //   component: () =>
+  //     import(/* webpackChunkName: "about" */ "../views/About.vue")
+  // } ?????? vat is this
```
Comment (george-iuriet) [inline src/router/index.js]: dinamically importing components, or lazy-loading, helps reducing the bundle size, as some routes will only load when needed. For example, you might not need a page that allows editing your profile if you are not logged in, therefore you don't load it. You can achieve that with
```
  component: () => import('../component/editProfilePicture.vue');
```

File: src/components/app-header.vue [hunk 18]
```diff
@@ -2,14 +2,14 @@
   <header>
     <v-toolbar flat>
       <router-link to="/">
-        <p>homefood</p>
+        <div class="display-2">homefood</div>
```
Comment (george-iuriet) [inline src/components/app-header.vue]: `display-2` is not used.

File: src/components/app-header.vue [hunk 19]
```diff
@@ -19,17 +19,17 @@
...
 </script>
 
 <!-- Add "scoped" attribute to limit CSS to this component only -->
 <style scoped lang="scss">
-p {
-  color: orange;
-  font-size: 170%;
-  font-weight: bold;
-}
+/*p {*/
```
Comment (george-iuriet) [inline src/components/app-header.vue]: I guess this can be removed

File: src/components/app-header.vue (hunk 11 above)
Comment (Valentin-Dumitrache) [reply src/components/app-header.vue]: Right, it's not, it was something I tried before I discovered v-spacer

File: src/components/app-header.vue (hunk 14 above)
Comment (Valentin-Dumitrache) [reply src/components/app-header.vue]: Yes, I noticed people don't like using them although I have pretty good cases of where they would be fit to be used. Anyways I'll stick to classes.

File: src/components/app-header.vue (hunk 18 above)
Comment (Valentin-Dumitrache) [reply src/components/app-header.vue]: Yes, I don't get why, I want to use it. This is regarding typography, let's discuss the whole subject tomorrow.

File: src/components/app-header.vue (hunk 19 above)
Comment (Valentin-Dumitrache) [reply src/components/app-header.vue]: Agreed, I was waiting on the class alternative to work

File: src/components/app-header.vue (hunk 15 above)
Comment (Valentin-Dumitrache) [reply src/components/app-header.vue]: Why? I'm removing text decorations for the router-link and styling it

File: src/router/index.js (hunk 17 above)
Comment (Valentin-Dumitrache) [reply src/router/index.js]: I see, I will keep that as a comment, just to be sure to remember to use it when needed. Let's discuss it further when we will need it.

File: src/App.vue (hunk 2 above)
Comment (Valentin-Dumitrache) [reply src/App.vue]: Done

File: src/components/app-header.vue (hunk 13 above)
Comment (Valentin-Dumitrache) [reply src/components/app-header.vue]: Not working for me, let's try to see why..

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (alina-dumitrache) [reply README.md]: This is still outstanding, I believe

File: src/router/index.js (hunk 17 above)
Comment (george-iuriet) [reply src/router/index.js]: then please refactor the comment to something like: 
```
component lazy-load:
component: () => import('../component/editProfilePicture.vue');
```
